test/test_transforms_v2_refactored.py

```diff
@@ -3009,3 +3013,104 @@ def test_auto_augment_policy_error(self):
+    def test_kernel(self, old_format, new_format, dtype, device):
+        check_kernel(
+            F.convert_bounding_box_format,
+            make_bounding_boxes(format=old_format, dtype=dtype, device=device),
+            new_format=new_format,
+            old_format=old_format,
+        )
+
+    @old_new_formats
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_inplace(self, old_format, new_format, dtype, device):
+        input = make_bounding_boxes(format=old_format, dtype=dtype, device=device).as_subclass(torch.Tensor)
+        input_version = input._version
+
+        if old_format == new_format:
+            output_inplace = F.convert_bounding_box_format(
+                input, old_format=old_format, new_format=new_format, inplace=True
+            )
+            assert output_inplace.data_ptr() == input.data_ptr()
+            assert output_inplace._version == input_version
+            assert output_inplace is input
+        else:
+            output_out_of_place = F.convert_bounding_box_format(input, old_format=old_format, new_format=new_format)
+            assert output_out_of_place.data_ptr() != input.data_ptr()
+            assert output_out_of_place is not input
+
+            output_inplace = F.convert_bounding_box_format(
+                input, old_format=old_format, new_format=new_format, inplace=True
+            )
+            assert output_inplace.data_ptr() == input.data_ptr()
+            assert output_inplace._version > input_version
+            assert output_inplace is input
+
+            assert_equal(output_inplace, output_out_of_place)
+
+    @old_new_formats
+    def test_functional(self, old_format, new_format):
+        check_functional(F.convert_bounding_box_format, make_bounding_boxes(format=old_format), new_format=new_format)
+
+    @old_new_formats
+    @pytest.mark.parametrize("format_type", ["enum", "str"])
+    def test_transform(self, old_format, new_format, format_type):
+        check_transform(
+            transforms.ConvertBoundingBoxFormat(new_format.name if format_type == "str" else new_format),
+            make_bounding_boxes(format=old_format),
+        )
+
+    def _reference_convert_bounding_box_format(self, bounding_boxes, new_format):
+        return tv_tensors.wrap(
+            torchvision.ops.box_convert(
+                bounding_boxes.as_subclass(torch.Tensor),
+                in_fmt=bounding_boxes.format.name.lower(),
+                out_fmt=new_format.name.lower(),
+            ).to(bounding_boxes.dtype),
+            like=bounding_boxes,
+            format=new_format,
+        )
+
+    @old_new_formats
+    @pytest.mark.parametrize("dtype", [torch.int64, torch.float32])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    @pytest.mark.parametrize("fn_type", ["functional", "transform"])
+    def test_correctness(self, old_format, new_format, dtype, device, fn_type):
```

Comment:
This seems to be the only test that needs to be parametrized over all format pairs, dtypes, and devices? Maybe we can simplify the other ones?